server/server.go

```diff
@@ -195,6 +197,7 @@ func (h *handler) loop(tconn net.Conn) {
 	for {
 		pkt := new(protocol.Packet)
 		<-h.tokens
+		h.conn.SetReadDeadline(time.Now().Add(h.timeout))
```

Comment:
Same as with `SetWriteDeadline`, error should be handled and we should break out of loop